### PR TITLE
Remove blob batch

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — APPROVED** [review]

File: sdk/storage/azure-storage-blobs/CHANGELOG.md [hunk 1]
```diff
@@ -65,6 +65,7 @@
 - Continuation token of result types are changed to nullable.
 - Rename `Models::DeleteSnapshotsOption::Only` to `Models::DeleteSnapshotsOption::OnlySnapshots`.
 - Rename `SourceConditions` in API options to `SourceAccessConditions`.
+- Remove Blob Batch.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/CHANGELOG.md:68]: Out of curiosity, what motivated the removal?

**Comment by ahsonkhan** [comment]
What about these things:
https://github.com/Azure/azure-sdk-for-cpp/blob/c3484c5d4daa38257c633f74cf86e44ed1bd68fc/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp#L188-L190
https://github.com/Azure/azure-sdk-for-cpp/blob/0643eeba5de821c0585223fdb160a3cf1a8fb1b0/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp#L655-L658

Please go through and make sure the removal of blob batch is complete.

**Comment by ahsonkhan** [comment]
@Jinming-Hu you marked these changes for the February milestone, is it meant for the next release rather than the upcoming Jan release?

**Comment by Jinming-Hu** [comment]
> @Jinming-Hu you marked these changes for the February milestone, is it meant for the next release rather than the upcoming Jan release?

Nah, it's a mistake.

**Comment by Jinming-Hu** [comment]
> What about these things:
> https://github.com/Azure/azure-sdk-for-cpp/blob/c3484c5d4daa38257c633f74cf86e44ed1bd68fc/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_service_client.hpp#L188-L190
> 
> 
> https://github.com/Azure/azure-sdk-for-cpp/blob/0643eeba5de821c0585223fdb160a3cf1a8fb1b0/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/protocol/blob_rest_client.hpp#L655-L658
> 
> I don't see any uses or calls to those types. How come it is being built successfully?
> 
> Please go through and make sure the removal of blob batch is complete.

Yeah, some of them were left out. I'm gonna fix it.

File: sdk/storage/azure-storage-blobs/CHANGELOG.md (hunk 1 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-blobs/CHANGELOG.md:68]: It will come back as a separate package as suggested by Ted.

**Comment by ahsonkhan** [comment]
> Nah, it's a mistake.

> Yeah, some of them were left out. I'm gonna fix it.

Sounds good 👍 
Let's prioritize getting this change in for the current release quickly :)

**Pull request merged**